packages/web-client/tests/unit/services/hub-authentication-test.ts

```diff
@@ -0,0 +1,234 @@
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
+import sinon from 'sinon';
+import HubAuthentication from '@cardstack/web-client/services/hub-authentication';
+import TestLayer2Web3Strategy from '@cardstack/web-client/utils/web3-strategies/test-layer2';
+import { settled } from '@ember/test-helpers';
+
+let HUB_AUTH_TOKEN = 'HUB_AUTH_TOKEN';
+let layer2AccountAddress = '0x182619c6Ea074C053eF3f1e1eF81Ec8De6Eb6E44';
+let layer2Service: TestLayer2Web3Strategy;
+let hubAuthentication: HubAuthentication;
+
+module('Unit | Service | HubAuthentication', function (hooks) {
+  setupTest(hooks);
+
+  hooks.beforeEach(function () {
+    // this is the condition for initializing with an authenticated state
+    // assumption made that layer2Service.checkHubAuthenticationValid returns Promise<true>
+    window.TEST__AUTH_TOKEN = HUB_AUTH_TOKEN;
+    layer2Service = this.owner.lookup('service:layer2-network').strategy;
+    layer2Service.test__simulateAccountsChanged([layer2AccountAddress]);
+  });
+
+  // Test initialization
+  test('it can initialize with an authenticated state', async function (assert) {
+    hubAuthentication = this.owner.lookup('service:hub-authentication');
```

Comment:
Does it work to move this into the `beforeEach`? I see it in the tests, but maybe there’s some subtle interaction that prevents that

Reply:
There are some cases where we want to set `window.TEST__AUTH_TOKEN` before we initialize a hub authentication service instance, so it is a bit awkward to move it into a `beforeEach`. I could organize stuff into modules (with hub auth instantiated vs without), though I am not sure if that's overkill

Reply:
ya I don’t love having to organise tests around setup requirements vs conceptual differences, and it doesn’t seem worth it to me for a one-line savings